OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/ide/instances/linear_constant_propagation/problem/LinearConstantPropagationProblem.scala

```diff
@@ -0,0 +1,641 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package tac
+package fpcf
+package analyses
+package ide
+package instances
+package linear_constant_propagation
+package problem
+
+import scala.annotation.unused
+
+import scala.collection
+import scala.collection.immutable
+
+import org.opalj.BinaryArithmeticOperators
+import org.opalj.ai.domain.l1.DefaultIntegerRangeValues
+import org.opalj.br.Method
+import org.opalj.fpcf.PropertyStore
+import org.opalj.ide.problem.EdgeFunctionResult
+import org.opalj.ide.problem.FlowFunction
+import org.opalj.ide.problem.IdentityFlowFunction
+import org.opalj.ide.problem.MeetLattice
+import org.opalj.tac.fpcf.analyses.ide.problem.JavaIDEProblem
+import org.opalj.tac.fpcf.analyses.ide.solver.JavaStatement
+import org.opalj.tac.fpcf.analyses.ide.solver.JavaStatement.StmtAsCall
+
+/**
+ * Definition of the linear constant propagation problem.
+ *
+ * @author Robin Körkemeier
+ */
+class LinearConstantPropagationProblem
+    extends JavaIDEProblem[LinearConstantPropagationFact, LinearConstantPropagationValue] {
+    override val nullFact: LinearConstantPropagationFact =
+        NullFact
+
+    override val lattice: MeetLattice[LinearConstantPropagationValue] =
+        LinearConstantPropagationLattice
+
+    override def getAdditionalSeeds(stmt: JavaStatement, callee: Method)(
+        implicit propertyStore: PropertyStore
+    ): collection.Set[LinearConstantPropagationFact] = {
+        callee.parameterTypes
```

Comment:
Consider placing a `.iterator` here to avoid repeatedly creating temporary collections